observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetryDeploymentProcessor.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.opentelemetry;
+
+import static org.jboss.as.weld.Capabilities.WELD_CAPABILITY_NAME;
+import static org.wildfly.extension.opentelemetry.OpenTelemetryExtensionLogger.OTEL_LOGGER;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.smallrye.opentelemetry.implementation.cdi.OpenTelemetryExtension;
+import org.jboss.as.controller.capability.CapabilityServiceSupport;
+import org.jboss.as.ee.structure.DeploymentType;
+import org.jboss.as.ee.structure.DeploymentTypeMarker;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.weld.WeldCapability;
+import org.wildfly.extension.opentelemetry.api.OpenTelemetryCdiExtension;
+import org.wildfly.extension.opentelemetry.api.WildFlyOpenTelemetryConfig;
+
+class OpenTelemetryDeploymentProcessor implements DeploymentUnitProcessor {
+    private final boolean useServerConfig;
+    private final WildFlyOpenTelemetryConfig serverConfig;
+
+    public OpenTelemetryDeploymentProcessor(boolean useServerConfig, WildFlyOpenTelemetryConfig serverConfig) {
+        this.useServerConfig = useServerConfig;
+        this.serverConfig = serverConfig;
+    }
+
+    @Override
+    public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
+        OTEL_LOGGER.processingDeployment();
```

Comment:
This can happen later, but we should remove < INFO logging from these logger interfaces and just inline the English text in a OTEL_LOGGER.debug.... call. Anything in those interfaces ends up being localized, which isn't free and isn't needed for < INFO. It's also confusing in the code to see the logger interfaces used for debug logging because the reader (e.g. me) is going to assume it's INFO or above.